### PR TITLE
Handles geckodriver/Marionette urls weirdness and untar driver

### PR DESCRIPTION
**Pull request by jucrouzet**
Fixes several problems with geckodriver/Marionette urls and files.

1. geckodriver was named wires before 0.8.0
2. The 0.3.0 version directory is names `/0.3.0/` instead of `/v0.3.0/` 
3. The file is sometimes names `[name]-v[version]-[arch]` and sometimes `[name]-[version]-[arch]`
4. Prior to 0.9.0, files were just gzipped, (not tar'd/gzipped)
5. .tar.gz files were not untar'd (extracted), resulting to : 

```
[...]
>> node_modules/selenium-standalone/.selenium/geckodriver/0.9.0-x64-geckodriver: 1: node_modules/selenium-standalone/.selenium/geckodriver/0.9.0-x64-geckodriver: Syntax error: ")" unexpected
```
(Which at the end throws a `org.openqa.selenium.remote.UnreachableBrowserException` as browser is not launched)

### CODE REVIEW

File: lib/install.js [hunk 1]
```diff
@@ -213,7 +215,31 @@ function install(opts, cb) {
     );
     fs.createReadStream(gzipFilePath).pipe(gunzip).pipe(writeStream)
       .once('error', cb.bind(null, new Error('Could not read ' + gzipFilePath)))
-      .once('finish', function() { cb() })
+      .once('finish', function() {
```
Comment (vvo) [inline lib/install.js]: Should'nt we check in advance that it's a tar.gz and avoid trying to ungzip etc?

File: lib/install.js [hunk 2]
```diff
@@ -213,7 +215,31 @@ function install(opts, cb) {
...
+        if (from.substr(-7) === '.tar.gz') {
+          var destDir = path.join(os.tmpdir(), 'selenium-standalone-' + Math.random());
```
Comment (vvo) [inline lib/install.js]: Can we do all in-memory instead of writing to disk? Maybe there's a module to handle this.

File: lib/install.js (hunk 1 above)
Comment (jucrouzet) [inline lib/install.js]: A tar.gz should be ungzipped before being untarred.

File: lib/install.js (hunk 2 above)
Comment (jucrouzet) [inline lib/install.js]: Probably, I will check for another module... Should be a good idea thought as it's a one file tar archive.
I make a commit this afternoon to correct it

File: lib/install.js (hunk 1 above)
Comment (vvo) [inline lib/install.js]: Right!

File: lib/install.js (hunk 2 above)
Comment (vvo) [inline lib/install.js]: Nice, let me know

**Comment by jucrouzet** [comment]
@vvo There you go, I used https://github.com/mafintosh/tar-stream which seems to do the job (tested both cases .tar.gz and .gz) : 

```
c2c@vbox-perso:selenium-standalone/ (master✗) $ ./bin/selenium-standalone install
----------
selenium-standalone installation starting
----------

---
selenium install:
from: https://selenium-release.storage.googleapis.com/2.53/selenium-server-standalone-2.53.1.jar
to: /home/c2c/work/icecast.js/node_modules/selenium-standalone/.selenium/selenium-server/2.53.1-server.jar
---
chrome install:
from: https://chromedriver.storage.googleapis.com/2.23/chromedriver_linux64.zip
to: /home/c2c/work/icecast.js/node_modules/selenium-standalone/.selenium/chromedriver/2.23-x64-chromedriver
---
firefox install:
from: https://github.com/mozilla/geckodriver/releases/download/v0.10.0/geckodriver-v0.10.0-linux64.tar.gz
to: /home/c2c/work/icecast.js/node_modules/selenium-standalone/.selenium/geckodriver/0.10.0-x64-geckodriver


selenium-standalone installation [=================   ] 84% 1.9s

-----
selenium-standalone installation finished
-----
c2c@vbox-perso:selenium-standalone/ (master✗) $ .selenium/geckodriver/0.10.0-x64-geckodriver -h
geckodriver 0.10.0

The source is available at https://github.com/mozilla/geckodriver

This program is subject to the terms of the Mozilla Public License 2.0.
You can obtain a copy of the license at https://mozilla.org/MPL/2.0/.
WebDriver implementation for Firefox.

USAGE:
    0.10.0-x64-geckodriver [FLAGS] [OPTIONS]

FLAGS:
        --connect-existing    Connect to an existing Firefox instance
    -h, --help                Prints help information
        --no-e10s             Start Firefox without multiprocess support (e10s) enabled
    -V, --version             Prints version information
    -v                        Set the level of verbosity. Pass once for debug level logging and twice for trace level logging

OPTIONS:
    -b, --binary <BINARY>           Path to the Firefox binary, if no binary capability provided
        --log <LEVEL>               Set Gecko log level [values: fatal, error, warn, info, config, debug, trace]
        --marionette-port <PORT>    Port to use to connect to gecko (default: random free port)
        --host <HOST>               Host ip to use for WebDriver server (default: 127.0.0.1)
    -p, --port <PORT>               Port to use for WebDriver server (default: 4444)

c2c@vbox-perso:selenium-standalone/ (master✗) $ ./bin/selenium-standalone install --drivers.firefox.version=0.7.1
----------
selenium-standalone installation starting
----------

---
selenium install:
from: https://selenium-release.storage.googleapis.com/2.53/selenium-server-standalone-2.53.1.jar
to: /home/c2c/work/icecast.js/node_modules/selenium-standalone/.selenium/selenium-server/2.53.1-server.jar
---
chrome install:
from: https://chromedriver.storage.googleapis.com/2.23/chromedriver_linux64.zip
to: /home/c2c/work/icecast.js/node_modules/selenium-standalone/.selenium/chromedriver/2.23-x64-chromedriver
---
firefox install:
from: https://github.com/mozilla/geckodriver/releases/download/v0.7.1/wires-0.7.1-linux64.gz
to: /home/c2c/work/icecast.js/node_modules/selenium-standalone/.selenium/geckodriver/0.7.1-x64-geckodriver
---
File from https://chromedriver.storage.googleapis.com/2.23/chromedriver_linux64.zip has already been downloaded
---
File from https://selenium-release.storage.googleapis.com/2.53/selenium-server-standalone-2.53.1.jar has already been downloaded


selenium-standalone installation [====================] 100% 0.0s


-----
selenium-standalone installation finished
-----
c2c@vbox-perso:selenium-standalone/ (master) $ .selenium/geckodriver/0.7.1-x64-geckodriver -h
Usage:
    .selenium/geckodriver/0.7.1-x64-geckodriver [OPTIONS]

WebDriver to marionette proxy.

optional arguments:
  -h,--help             show this help message and exit
  -b,--binary BINARY    Path to the Firefox binary
  --webdriver-host WEBDRIVER_HOST
                        Host to run webdriver server on
  --webdriver-port WEBDRIVER_PORT
                        Port to run webdriver on
  --marionette-port MARIONETTE_PORT
                        Port to run marionette on
  --connect-existing    Connect to an existing firefox process
  --e10s                Load Firefox with an e10s profile

```

Let me know if you want me to rebase

**Comment by vvo** [comment]
awesome! I am adding you as a contributor too :)

**Comment by vvo** [comment]
released

**Pull request merged**